public/javascripts/lti/messages.js

```diff
@@ -0,0 +1,74 @@
+import $ from 'jquery'
+
+var beforeUnloadHandler;
+function setUnloadMessage(msg) {
+  removeUnloadMessage();
+
+  beforeUnloadHandler = function(e) {
+    return (e.returnValue = msg || "");
+  }
+  window.addEventListener('beforeunload', beforeUnloadHandler);
+}
+
+function removeUnloadMessage() {
+  if (beforeUnloadHandler) {
+    window.removeEventListener('beforeunload', beforeUnloadHandler);
+    beforeUnloadHandler = null;
+  }
+}
+
+function findDomForWindow(sourceWindow) {
+  const iframes = document.getElementsByTagName("IFRAME");
+  for (let i=0; i < iframes.length; i += 1) {
+    if (iframes[i].contentWindow === sourceWindow) {
+      return iframes[i];
+    }
+  }
+  return null;
+}
+
+export function monitorLtiMessages() {
+  window.addEventListener('message', function(e) {
+    try {
+      var message = JSON.parse(e.data);
+      switch (message.subject) {
+        case 'lti.frameResize':
+          let height = message.height;
+          if (height <= 0) height = 1;
+          const iframe = findDomForWindow(e.source);
+          if (typeof height === 'number') {
+            height = height + 'px';
+          }
+          iframe.height = height;
```

Comment:
this was tested out with an LTI tool that does post messages but resizes failed. I believe we need to resize the container around the iframe, not the iframe itself.